pkg/hydrate/controller.go

```diff
@@ -115,13 +115,32 @@ func (h *Hydrator) runHydrate(sourceCommit, syncDir string) HydrationError {
 	if err := kustomizeBuild(syncDir, dest, true); err != nil {
 		return err
 	}
+
+	changed, commit, err := h.checkCommitChanged(sourceCommit)
+	if err != nil {
+		return NewTransientError(err)
```

Comment:
ditto here regarding whether this is a transient error

Reply:
Simplified the nested func call so error should be transient.